server/src/main/java/org/elasticsearch/cluster/ClusterStateTaskExecutor.java

```diff
@@ -55,15 +56,7 @@ default void clusterStatePublished(ClusterChangedEvent clusterChangedEvent) {
      * This allows groupd task description but the submitting source.
      */
     default String describeTasks(List<T> tasks) {
-        return tasks.stream().map(T::toString).reduce((s1,s2) -> {
-            if (s1.isEmpty()) {
-                return s2;
-            } else if (s2.isEmpty()) {
-                return s1;
-            } else {
-                return s1 + ", " + s2;
-            }
-        }).orElse("");
+        return String.join(", ", tasks.stream().map(t -> (CharSequence)t.toString()).filter(t -> t.length() == 0)::iterator);
```

Comment:
out of curiosity - didn't `T::toString` work?

Reply:
No, apparently `Iterator<String> cannot be converted to Iterator<CharSequence>`:

```
> Task :server:compileJava 
/Users/davidturner/src/elasticsearch-master/server/src/main/java/org/elasticsearch/cluster/ClusterStateTaskExecutor.java:59: error: no suitable method found for join(String,tasks.stre[...]rator)
        return String.join(", ", tasks.stream().map(T::toString).filter(t -> t.length() == 0)::iterator);
                     ^
    method String.join(CharSequence,CharSequence...) is not applicable
      (varargs mismatch; CharSequence is not a functional interface
          multiple non-overriding abstract methods found in interface CharSequence)
    method String.join(CharSequence,Iterable<? extends CharSequence>) is not applicable
      (argument mismatch; bad return type in method reference
          Iterator<String> cannot be converted to Iterator<CharSequence>)
```